corehq/apps/app_manager/static/app_manager/js/app_manager.js

```diff
@@ -232,12 +232,159 @@ hqDefine('app_manager/js/app_manager', function () {
+                var parent = modulesByUid[$(childModule).data('rootmoduleid')];
+                if (!parent) {
+                    // This child module is orphaned, throw it at the end and alert user
+                    $("ul.appnav-module").append(childModule);
+                    var moduleName = $(childModule).data('modulename'),
+                        alertText = _.template(gettext(
+                            'Menu "<%= moduleName %>" references a parent menu that no longer exists'
+                        ))({moduleName: moduleName});
+                    hqImport('hqwebapp/js/alert_user').alert_user(alertText, "danger");
+                } else {
+                    addChildModuleToParent(childModule, parent);
+                }
+            });
+        }
+        function addChildModuleToParent(childModule, parent) {
+            var childList = $(parent).find("ul.child-modules");
+            if (childList.length === 0) {
+                childList = $('<ul class="appnav-menu child-modules sortable"></ul>');
+                $(parent).append(childList);
+            }
+            childList.append(childModule);
+        }
+        function modulesWereReordered() {
+            return _.some($(".module"), function (element, index) {
+                return index !== $(element).data('index');
+            });
+        }
+        function promptToSaveOrdering() {
+            $("#reorder_modules_modal").modal('show');
+        }
         function updateRelatedTags($elem, name, value) {
             var relatedTags = $elem.find("[data-" + name + "]");
             _.each(relatedTags, function (related) {
                 $(related).data(name, value);
             });
         }
+        function initSortable($sortable) {
+            var options = {
+                handle: '.drag_handle ',
+                items: ">*:not(.sort-disabled)",
+                update: function (e, ui) { updateAfterMove(e, ui, $sortable); },
+            };
+            if ($sortable.hasClass('sortable-forms')) {
+                options["connectWith"] = '.sortable-forms';
+            }
+            $sortable.sortable(options);
+        }
+        function updateAfterMove(e, ui, $sortable) {
+            // because the event is triggered on both sortables when moving between one sortable list to
+            // another, do a check to see if this is the sortable list we're moving the item to
+            if ($sortable.find(ui.item).length < 1) { return; }
+
+            var toModuleUid = $sortable.parents('.edit-module-li').data('uid'),
+                fromModuleUid = ui.item.data('moduleuid'),
+                sortingForms = $sortable.hasClass('sortable-forms'),
+                movingFormToNewModule = sortingForms && toModuleUid !== fromModuleUid;
+
+            var from, to;
+            if (movingFormToNewModule) {
+                [from, to] = calculateMoveFormToNewModule($sortable, ui, toModuleUid);
```

Comment:
Huh, I haven't seen that style in js. I like the style from a coding perspective, but I'm opposed to parse errors, since they can make it harder to integrate with other tools that expect correct syntax. We've had issues with the uglifier failing on js syntax that's illegal but accepted by browsers, specifically having a trailing comma in a list of function args.

I'm curious, does the uglier work properly on this? You can test by running `build_requirejs` locally (make sure to `rm -rf staticfiles` and run `collectstatic` and `compilejsi18n` first).

Reply:
Dang, I'll just switch it to basic JS, how's this look?
```js
var move;
if ... {
    move = ...;
} else {
    move = ...;
}
var from = move[0],
    to = move[1]
```